labdao/lab-exchange
# Wallet-based gating of gateway

## What type of PR is this? (check all applicable)


## Description

This PR adds wallet gating to the gateway app. User membership is determined by if their wallet address holds a Record, which is determined by hitting our Alchemy RPC endpoint. Access to various gateway endpoints requires member status set to true for the wallet address.

## Steps to Test

1. Log in with a wallet which holds a Record.
2. Check if Membership Status is correctly listed in the `TopNav` component.
3. Add a flow.
4. Log out.
5. Log in with a wallet which does not hold a Record.
6. Confirm frontend fetches the correct member status state by checking if Membership Status is False in the `TopNav` component.
7. Attempt to create a Flow with non-member wallet. Expected behavior is to have request rejected by backend.

## Relevant GIF 

<img src="https://media0.giphy.com/media/QvSGhHq8CrVzq/giphy.gif"/>
